main.py

```diff
@@ -24,6 +27,10 @@ def welcome():
         verify_error= "Passwords don't match!"
         return render_template('index.html', verify_error=verify_error)
     
+    elif not re.match(pattern, email):
+        email_error = "Invalid email"
+        return render_template('index.html', email_error=email_error)
+
```

Comment:
Made another `elif` to check if the email is valid. `re.match()` matches the regular expression pattern against the email submitted. An error should return if it *doesn't* match, so it is set to `elif not`.